### PR TITLE
fix(ui-ux): Fix bugs on composite swap screen v2

### PR DESCRIPTION
**Pull request by G9000**
Fixing the following bug on the composite swap screen:

- Missing line break on Want token row
- Dynamic dToken message
- Border radius on token selection and confirm swap screen header
- Fix want token border issue
- Fix missing amount to receive on signer

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx [hunk 1]
```diff
@@ -553,12 +553,12 @@ export function CompositeSwapScreenV2({ route }: Props): JSX.Element {
       if (selectedTokenA.displaySymbol === "DUSD") {
         message = translate(
           "screens/CompositeSwapScreen",
-          "You are buying dtokens at 5% more than the oracle price at settlement block"
+          "You are buying dTokens at 5% more than the oracle price at settlement block"
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx:556]:
```suggestion

          "You are buying {{displaySymbol}} at 5% more than the oracle price at settlement block",
          {
            displaySymbol: selectedTokenA.displaySymbol,
          }
```

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx [hunk 2]
```diff
@@ -553,12 +553,12 @@ export function CompositeSwapScreenV2({ route }: Props): JSX.Element {
...
         );
       } else {
         message = translate(
           "screens/CompositeSwapScreen",
-          "You are selling your dtoken at 5% less than the oracle price at settlement block"
+          "You are selling your dToken at 5% less than the oracle price at settlement block"
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx:561]:
```suggestion
          "You are selling your {{displaySymbol}} at 5% less than the oracle price at settlement block",
          {
            displaySymbol: selectedTokenA.displaySymbol,
          }
```

File: shared/translations/languages/de.json [hunk 3]
```diff
@@ -1012,8 +1012,8 @@
     "Are you certain you want to proceed with this swap despite the DEX Stabilization fees of {{fee}} that will be incurred as part of the composite path (DUSD -> {{tokenB}})?": "Bist du dir sicher, dass du diesen Tausch trotz der DEX-Stabilisierungsgebühren in Höhe von {{fee}}, die im Verlauf des zusammengesetzten Pfads beim Komposit-Swap (DUSD -> {{tokenB}}) anfallen werden, durchführen willst?",
     "DEX stabilization fee": "DEX-Stabilisierungsgebühr",
     "Future swap uses the oracle price of the selected token on the settlement block": "Future Swap verwendet den Orakelpreis des ausgewählten Tokens beim Abrechnungsblock",
-    "You are buying dtokens at 5% more than the oracle price at settlement block": "Du kaufst dToken zu einem Preis, der 5% über dem Orakelpreis beim Abrechnungsblock liegt",
-    "You are selling your dtoken at 5% less than the oracle price at settlement block": "Du verkaufst dToken zu einem Preis, der 5% unter dem Orakelpreis beim Abrechnungsblock liegt",
+    "You are buying dTokens at 5% more than the oracle price at settlement block": "Du kaufst dToken zu einem Preis, der 5% über dem Orakelpreis beim Abrechnungsblock liegt",
```
Comment (pierregee) [inline shared/translations/languages/de.json:1015]: Also need this to be dynamic ^^

File: shared/translations/languages/de.json [hunk 4]
```diff
@@ -1012,8 +1012,8 @@
...
+    "You are selling your dToken at 5% less than the oracle price at settlement block": "Du verkaufst dToken zu einem Preis, der 5% unter dem Orakelpreis beim Abrechnungsblock liegt",
```
Comment (pierregee) [inline shared/translations/languages/de.json:1016]: same here

**Review by pierregee — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx [hunk 5]
```diff
@@ -573,12 +573,18 @@ export function CompositeSwapScreenV2({ route }: Props): JSX.Element {
       if (selectedTokenA.displaySymbol === "DUSD") {
         message = translate(
           "screens/CompositeSwapScreen",
-          "You are buying dtokens at 5% more than the oracle price at settlement block"
+          "You are buying {{displaySymbol}} at 5% more than the oracle price at settlement block",
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx:576]: The token passed is wrong. It should be tokenB

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: shared/translations/languages/zh-Hant.json [hunk 6]
```diff
@@ -1016,8 +1016,9 @@
     "Are you certain you want to proceed with this swap despite the DEX Stabilization fees of {{fee}} that will be incurred as part of the composite path (DUSD -> {{tokenB}})?": "Are you certain you want to proceed with this swap despite the DEX Stabilization fees of {{fee}} that will be incurred as part of the composite path (DUSD -> {{tokenB}})?",
     "DEX stabilization fee": "DEX stabilization fee",
     "Future swap uses the oracle price of the selected token on the settlement block": "Future swap uses the oracle price of the selected token on the settlement block",
-    "You are buying dtokens at 5% more than the oracle price at settlement block": "You are buying dtokens at 5% more than the oracle price at settlement block",
-    "You are selling your dtoken at 5% less than the oracle price at settlement block": "You are selling your dtoken at 5% less than the oracle price at settlement block",
+    "You are buying dTokens at 5% more than the oracle price at settlement block.": "You are buying dTokens at 5% more than the oracle price at settlement block.",
```
Comment (pierregee) [inline shared/translations/languages/zh-Hant.json:1019]:
```suggestion
```

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx [hunk 7]
```diff
@@ -571,19 +571,30 @@ export function CompositeSwapScreenV2({ route }: Props): JSX.Element {
...
     if (selectedTokenA !== undefined) {
       if (selectedTokenA.displaySymbol === "DUSD") {
-        message = translate(
-          "screens/CompositeSwapScreen",
-          "You are buying dtokens at 5% more than the oracle price at settlement block"
-        );
+        if (selectedTokenB === undefined) {
+          message = translate(
+            "screens/CompositeSwapScreen",
+            "You are buying dTokens at 5% more than the oracle price at settlement block."
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx:577]: Hey, missed this one, need to update this to dynamic tokens

File: shared/translations/languages/zh-Hant.json (hunk 6 above)
Comment (pierregee) [reply shared/translations/languages/zh-Hant.json:1019]: Can remove for all translations

**Review by pierregee — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx (hunk 7 above)
Comment (G9000) [reply mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx:577]: but when tokenB empty? wont it make sense just let it to dTokens?

Comment (pierregee) [reply mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/CompositeSwapScreenV2.tsx:577]: right, thought it is tokenA

**Comment by pierregee** [comment]
image
Border radius on Portfolio -> Swap -> Token Selection Screen

File: mobile-app/app/screens/AppNavigator/screens/Dex/DexNavigator.tsx [hunk 8]
```diff
@@ -304,14 +304,6 @@ export function DexNavigator(): JSX.Element {
...
             <HeaderNetworkStatus onPress={goToNetworkSelect} />
           ),
-          headerStyle: [
-            screenOptions.headerStyle,
-            tailwind("rounded-b-none border-b-0"),
-            {
-              shadowOpacity: 0,
-              height: (Platform.OS !== "android" ? 88 : 96) + insets.top,
-            },
-          ],
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/DexNavigator.tsx:314]: can we keep everything else? The UI alignment is visible when transitioning
```  headerStyle: [
            screenOptions.headerStyle,
            {
              shadowOpacity: 0,
              height: (Platform.OS !== "android" ? 88 : 96) + insets.top,
            },
          ],

**Review by pierregee — APPROVED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**